doc/rfcs/inactive_users_cleanup.md

```diff
@@ -0,0 +1,45 @@
+# Inactive user cleanup
+When user registration finishes and users not passing "activation", corresponding DB record stays for ttl.
+But all metadata/alias/email records stay forever.
+
+1. Add additional Redis HASH and ZSET handling inactive users list.
+2. Change registration action:
+    - remove ttl assignment from from `USER_DATA` key
+    - call inactive list append process `addInactiveUser` 
+3. Change activation action:
+    - add inactive list cleanup `deleteInactiveUser`
+4. Add interval task.
+5. Add config parameter
+6. Move 'delete user' transaction function into separate file from `actions/remove.js` to `utils/removeUser.js`
+
+## Config changes
+Added new `deleteInactiveAccountsInterval` seconds to run cleaner task.
+
+## Constants
+New constants `USERS_ACTIVATE` - ZSET handling userId regtime, `USERS_ACTIVATE_AUDIENCE` - userId to audience binding
+
+## utils/inactiveUsers.js
+Handles all logic, including task creator, and database actions for InactiveUsersList
+
+### addInactiveUser(id, audience)
+Adds corresponding id record into `USERS_ACTIVATE` with score eq current time. Sets `USERS_ACTIVATE_AUDIENCE` hash to track audieces provided in registration process
+
```

Comment:
User may have multiple audiences associated with it and right now its one of the problems during removal of such users. We may need to add tracking of used audiences as well when updateMetadata methods are used so that we have references to them and cleanup the data in them as well

Reply:
https://github.com/makeomatic/ms-users/commit/0dc98f24365d9c5337993c781515e74eeb37ef3b audience tracking  and cleanup already in commit. Possible solution, describes pre logic. Current logic in pre text